### PR TITLE
Add migration for rm user default projector.

### PR DESCRIPTION
**Pull request by reiterl**
Add test.

### CODE REVIEW

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from datastore.migrations import RemoveFieldsMigration
+
+
+class Migration(RemoveFieldsMigration):
+    """
+    This migration removes the field `meeting/default_projector_$user_ids`.
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:6]: Ah, I forgot: `user` must also be removed from `meeting/default_projector_$_ids`, if present. Please extend this migration by overwriting the `migrate` method.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:16]: Done.

**Comment by reiterl** [comment]
The CI makes problems. Can someone look at it?

**Comment by reiterl** [comment]
Ubunutu 18.04 is deprecated and will be removed on 1st April 2023 . Today is a brownout day. Error msg:
Build and test production docker image
This is a scheduled Ubuntu-18.04 brownout. The Ubuntu-18.04 environment is deprecated and will be removed on April 1st, 2023. For more details, see https://github.com/actions/runner-images/issues/6002

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        for field in ("default_projector_$_ids", "used_as_default_$_in_meeting_id"):
+            if obj.get(field) and "user" in obj[field]:
+                obj[field].remove("user")
+                if not obj[field]:
+                    del obj[field]
+
+    def migrate_event(
+        self,
+        event: BaseEvent,
+    ) -> Optional[List[BaseEvent]]:
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:40]: You can inherit from `RemoveFieldsMigration` and call `super().migrate_event(event)` here. Then, assert that the return value only contains one event and execute your `remove_replacement` code on that. This removes the code duplication.

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        "projector": ["used_as_default_$user_in_meeting_id"],
+    }
+
+    def remove_fields(self, obj: Dict[str, Any], fields: List[str]) -> None:
+        for field in fields:
+            if field in obj:
+                del obj[field]
+
+    def remove_replacement(self, obj: Dict[str, Any]) -> None:
+        for field in ("default_projector_$_ids", "used_as_default_$_in_meeting_id"):
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:31]: You can pass the collection to this method adn then iterate over `collection_fields_map[collection] instead. Removes unnecessary iterations and duplication fo the field names.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:31]: Nope, the fields in collection_fields_map are different.

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:40]: Done.

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py]: I used a collection_replacement_fields_map for the other fields.

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py [hunk 4]
```diff
@@ -38,23 +38,17 @@ def migrate_event(
         self,
         event: BaseEvent,
     ) -> Optional[List[BaseEvent]]:
+        events = super().migrate_event(event)
+        if events is None:
+            return events
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:43]: This should never happen, but you should nevertheless still use the old event in this case to fully execute the migration.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py (hunk 3 above)
Comment (jsangmeister) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py]: Ah right, that's fine then.

File: openslides_backend/migrations/migrations/0038_remove_user_default_projector.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:43]: As far as I understand the code of RemoveFieldsMigration this case can happen, if collection doesn't match with one collection in the dict.

Comment (reiterl) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:43]: Then I want to stop processing anymore.

Comment (jsangmeister) [reply openslides_backend/migrations/migrations/0038_remove_user_default_projector.py:43]: Right! My bad

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**